nifi-nar-bundles/nifi-aws-bundle/nifi-aws-abstract-processors/src/main/java/org/apache/nifi/processors/aws/AwsClientProvider.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.aws;
+
+import com.amazonaws.AmazonWebServiceClient;
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import org.apache.nifi.processor.ProcessContext;
+
+public interface AwsClientProvider<ClientType extends AmazonWebServiceClient> {
+    /**
+     * Create client from the arguments
+     * @param context process context
+     * @param credentials static AWS credentials
+     * @param config AWS client configuration
+     * @return ClientType AWS client
+     *
+     * @deprecated use {@link #createClient(ProcessContext, AWSCredentialsProvider, ClientConfiguration) }
+     */
+    @Deprecated
+    ClientType createClient(final ProcessContext context, final AWSCredentials credentials, final ClientConfiguration config);
+
+    /**
+     * Create AWS client using credentials provider. This is the preferred method for creating AWS clients
+     * @param context process context
+     * @param credentialsProvider AWS credentials provider
+     * @param config AWS client configuration
+     * @return ClientType AWS client
+     */
+    ClientType createClient(final ProcessContext context, final AWSCredentialsProvider credentialsProvider, final ClientConfiguration config);
```

Comment:
With moving the cache access logic to `getClient()`, this `AwsClientProvider` interface can expose `createClient(ProcessContext, AwsClientDetails)` only, which is the "original" / "main" method for creating the client. In this case we can get rid of `AWSCredentials`, `AWSCredentialsProvider` and `ClientConfiguration` in this interface (those are rather implementation details in the abstract processors).